samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/translation/MicrosoftTranslator.java

```diff
@@ -54,15 +54,17 @@ public MicrosoftTranslator(Configuration configuration) {
         return CompletableFuture.supplyAsync(() -> {
             // From Cognitive Services translation documentation:
             // https://docs.microsoft.com/en-us/azure/cognitive-services/Translator/quickstart-translator?tabs=java
-            String json = String.format("[{ \"Text\": \"%s\" }]", text);
+            String body = String.format("[{ \"Text\": \"%s\" }]", text);
 
-            String uri =
-                MicrosoftTranslator.HOST + MicrosoftTranslator.PATH + MicrosoftTranslator.URI_PARAMS + targetLocale;
+            StringBuilder uri = new StringBuilder(MicrosoftTranslator.HOST)
+                .append(MicrosoftTranslator.PATH)
+                .append(MicrosoftTranslator.URI_PARAMS)
+                .append(targetLocale);
 
-            RequestBody requestBody = RequestBody.create(MediaType.parse("application/json"), json);
+            RequestBody requestBody = RequestBody.create(MediaType.parse("application/json"), body);
 
             Request request = new Request.Builder()
-                .url(uri)
+                .url(uri.toString())
```

Comment:
This `toString()` could be added in the definition of the uri variable, instead of use it here

Reply:
I like that, and so can be done in the catch block

Reply:
Updated!